src/python/pants/backend/python/target_types.py

```diff
@@ -144,6 +144,31 @@ def normalized_value(self, python_setup: PythonSetup) -> str:
         return resolve
 
 
+class PythonRunGoalUseSandboxField(BoolField):
+    alias = "run_goal_use_sandbox"
+    # ...
+    default = False
```

Comment:
As far as I can tell, this isn't a breaking change because:
- If the user was using the filename thats owned by both a `python_source` and `pex_binary`, we use the binary and this code isn't considered (and the binary uses sandbox-only)
- Otherwise they're using the `python_source`, so this is a new feature